src/App.js

```diff
@@ -1,5 +1,135 @@
+import { Console } from "@woowacourse/mission-utils";
+import InputView from "./InputView.js";
+import Validation from "./validation.js";
+import OutputView from "./OutputView.js";
+import { MENU } from "./constants/menu.js";
+import { EVENT, EVENT_DAYS, OFFER_MENU } from "./constants/event.js";
+
 class App {
-  async run() {}
+  /** 방문 예정 날짜 @type {number} */
+  #date;
+  /** 주문 메뉴 @type {Map<string, string>} */
+  #menusMap;
+
+  #totalAmount = 0;
+  #discount = { christmas: 0, weekdays: 0, weekends: 0, special: false, isOffer: false };
+  #totalDiscountAmount = 0;
+  #finalAmount = 0;
+
+  async run() {
+    this.#date = await this.#getDate();
+    this.#menusMap = new Map(await this.#getMenus());
+    OutputView.printDate(this.#date);
+    OutputView.printMenu(this.#menusMap);
+    this.#totalAmount = this.#calculateTotalAmount();
+    OutputView.printTotalAmount(this.#totalAmount);
+    this.#discount.isOffer = this.#isOffer();
+    OutputView.printOffer(this.#discount.isOffer);
+    this.#getDiscount();
+    OutputView.printDiscount(this.#discount);
+    this.#totalDiscountAmount = this.#calculateTotalDiscountAmount();
+    OutputView.printTotalDiscountAmount(this.#totalDiscountAmount);
+    this.#finalAmount = this.#calculateFinalAmount();
+    OutputView.printFinalAmount(this.#finalAmount);
+    OutputView.printEventBadge(this.#totalDiscountAmount);
+  }
+
+  /** 날짜를 입력받고 this.#date에 날짜를 number형으로 저장합니다. */
+  async #getDate() {
+    try {
+      const date = await InputView.readDate();
+      Validation.date(date);
+      return Number(date);
+    } catch (error) {
+      Console.print(error.message);
+      await this.#getDate();
+    }
+  }
+
+  /** 주문할 메뉴를 입력받고 this.#menusMap에 메뉴를 Map<메뉴, 개수> 형식으로 저장합니다. */
+  async #getMenus() {
+    try {
+      const menus = await InputView.readMenus();
+      Validation.menus(menus);
+      return menus.map((menu) => menu.split("-"));
+    } catch (error) {
+      Console.print(error.message);
+      await this.#getMenus();
+    }
+  }
+
+  /** 주문 받은 메뉴를 기반으로 할인 전 총 주문금액을 구합니다. */
+  #calculateTotalAmount() {
+    // 메뉴판의 모든 메뉴와 가격을 객체 형태로 새롭게 생성
+    const allMenuPrice = {};
+    Object.values(MENU).forEach((category) =>
+      Object.entries(category).forEach(([_i, { name, price }]) => (allMenuPrice[name] = price))
+    );
+
+    let totalAmount = 0;
+    this.#menusMap.forEach((count, dish) => {
+      totalAmount += allMenuPrice[dish] * count;
+    });
+    return totalAmount;
```

Comment:
`reduce` 한번 검색해보시고 사용해보시는 것도 좋을 것 같아요! 

물론 `forEach`로 결과를 만들 수 있지만 지금 목적에서는 `reduce`가 조금 더 어울리는 것 같네요🙃

Reply:
아직 그런 메서드 사용이 어색하네요ㅠㅠ 존재를 알고 있어도 막상 사용하려면 헷갈리는.. 좀더 연습해보겠습니다! 감사합니다 👍

Reply:
여기선 `Map`객체여서 `reduce`사용이 어렵겠네요ㅠㅠ 그래도 좋은 조언 감사합니당!

Reply:
```suggestion
    return Array.from(menusMap.entries()).reduce((acc, [dish, count]) => {
      return acc + allMenuPrice[dish] * count;
    }, 0);
```

이렇게는 수정 가능한 것 같긴 합니다! 👍